File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/RentOffer/UserRentOfferResponse.cs
using ByteBuy.Core.DTO.Public.Image;
using ByteBuy.Core.DTO.Public.Money;
using ByteBuy.Core.DTO.Public.Offer.Common;

namespace ByteBuy.Core.DTO.Public.Offer.RentOffer;

public record UserRentOfferResponse(
   Guid Id,
   Guid CategoryId,
   Guid ConditionId,
   string Name,
   string Description,
   int QuantityAvailable,
   OfferAddressResponse OfferAddress,
   MoneyDto PricePerDay,
   int MaxRentalDays,
   IReadOnlyCollection<ImageResponse> Images,
   IReadOnlyCollection<Guid>? ParcelLockerDeliveries,
   IReadOnlyCollection<Guid> OtherDeliveriesIds);

File: Byte&Buy.AngularSPA/src/app/core/dto/offers/rent/user-rent-offer-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../common/money-dto";
import { ImageResponse } from "../../image/image-response";
import { OfferAddressResponse } from "../common/offer-address-response";

export interface UserRentOfferResponse{
    id: Guid;
    categoryId: Guid;
    conditionId: Guid;
    name: string;
    description: string;
    quantityAvailable: number;
    pricePerDay: MoneyDto;
    offerAddress: OfferAddressResponse;
    maxRentalDays: number;
    images: ImageResponse[];
    parcelLockerDeliveries: Guid[] | null;
    otherDeliveriesIds: Guid[];
}